src/router.ts

```diff
@@ -325,6 +327,26 @@ export class OpenAPIRouter<D extends Document = Document> {
               const commaParsed = parseQuery(commaQueryString, { comma: true });
               query[queryParam] = commaParsed[queryParam];
             }
+            const schemaType = (parameter.schema as PickVersionElement<D, OpenAPIV3.SchemaObject, OpenAPIV3_1.SchemaObject>)?.type;
+            if (Array.isArray(query[queryParam])) {
+              query[queryParam] = (query[queryParam] as any[]).map((par) => {
+                // coerce the input to the specified type
+                if (schemaType === 'boolean') {
+                  return par === 'false' ? false : Boolean(par);
```

Comment:
I'm thinking we need a more robust way to parse booleans from strings.

```
Boolean('0') // false
```

Reply:
I didn't know about [ajv type coercion](https://ajv.js.org/coercion.html) when I first did this but I found out about it so I changed this to use that instead.